src/app/app.component.ts

```diff
@@ -47,19 +47,21 @@ export class AppComponent implements AfterViewInit {
   selectedNodeUrl?: string;
 
   // the folder the user has selected in the folder-browser
-  selectedFolder?: LfFolder;
-
+  selectedFolderPath: string;
+  selectedFolderId: number;
```

Comment:
If we actually need to save all of these, maybe we should have an object with all of them since they have the same lifecycle